tests/test_swapped_models.py

```diff
@@ -0,0 +1,52 @@
+import importlib
+
+from django.db import migrations
+from django.test import TestCase
+from django.test.utils import override_settings
+
+
+class TestSwappableModels(TestCase):
+    """
+    This test relies entirely on the `test_project` app and the following instruction:
+
+    python -m django makemigrations test_project --settings=tests.test_project.settings
+
+    This instruction is executed in travis.yml, and if you want to execute this test locally
+    you'll need to run it by yourself.
+    """
+
+    def _get_migration_class(self):
+        try:
+            module = importlib.import_module('tests.test_project.migrations.0001_initial')
+            return module.Migration
+        except ImportError:
+            raise RuntimeError(self.__doc__.strip())
+
+    @override_settings(SCHEDULER_CALENDAR_MODEL='schedule.Calendar', SCHEDULER_EVENT_MODEL='schedule.Event')
```

Comment:
Why is this overridden to not be the alternate model? Shouldn't the new tests use the `CustomEvent` class?

Reply:
I had an error with it but it might be related to my migration issue

Reply:
ah I know why:
- the tests are executed within the context of the `tests` project, using the `tests.settings` module
- the migrations is meaningful only in the context of the `tests.test_project`, using the `tests.test_project.settings`
- as a consequence, importing the migration file like I do in the tests results in an import error: the `django.conf.settings.SCHEDULER_CALENDAR_MODEL` isn't defined for `tests.settings` but it is defined for `tests.test_project.settings`